instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/OpenTelemetrySupplier.java

```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.spring.autoconfigure;
+
+import io.opentelemetry.api.OpenTelemetry;
+import java.util.function.Supplier;
+
+/** To inject an OpenTelemetry into bean post processors */
+public interface OpenTelemetrySupplier extends Supplier<OpenTelemetry> {}
```

Comment:
Is the addition of an interface really needed?

Reply:
Should the comment specify that the class is for internal use?

Reply:
The interface name should perhaps mentioned that this interface is only for post processor usage. `OpentTelemerySupplierInPostProcessor`?

Reply:
I moved it to internal package, but left the name as is. 

> Is the addition of an interface really needed?

Do you have a better idea how to do this? To me using a separate interface felt like the simplest approach to ensure that there is only 1 bean that matches.

Reply:
I agree. By declaring a bean of type `Supplier<OpenTelemetry>`, the user could also declare in his code this type of bean and the application will fail to start.